### PR TITLE
Closes #1845 fix create period dataset bug

### LINKED ISSUE #1845: Bug: create_period_dataset when both DT and DTM vars exist and DT requested
### What happened?

When trying to create PERIOD_REF dataset from ADSL period variables using `create_period()` function, an ERROR is produced when using `new_vars = exprs(APERSDT = APxxSDT, APEREDT = APxxEDT)` and where APxxSDT and APxxSDTM exist in the input ADSL dataset. The code searches for the ADSL variables contained in `new_vars` ie. `APxxSDT` and `APxxEDT`. The use of function `matches` picks up both `APxxSDT` and `APxxSDTM` as well as `APxxEDT` and `APxxEDTM`, which triggers an ERROR later about not having unique observations.

### Session Information

_No response_

### Reproducible Example

 ```
create_period_dataset(
   adsl,
   new_vars = exprs(APERSDT = APxxSDT, APEREDT = APxxEDT, TRTA = TRTxxA)
)
```

Code above is from header of function, the issue is when for example `AP01SDT` and `AP01SDTM` exist, it will find both of them, instead of just `AP01SDT`

### PR DESCRIPTION
**Pull request by zdz2101**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by zdz2101** [comment]
@millerg23 can you try rerunning your code but with this branch installed pertaining to: 

Code above is from header of function, the issue is when for example AP01SDT and AP01SDTM exist, it will find both of them, instead of just AP01SDT

File: R/period_dataset.R [hunk 1]
```diff
@@ -187,8 +187,8 @@ create_period_dataset <- function(dataset,
     }
     if (mode == "subperiod") {
       period_ref[[i]] <- pivot_longer(
-        select(dataset, !!!subject_keys, matches(cols[[i]])),
-        matches(cols[[i]]),
+        select(dataset, !!!subject_keys, matches(paste0(cols[[i]], "\\b", sep = ""))),
```
Comment (millerg23) [inline R/period_dataset.R:190]: Do we need `sep = ""` with `paste0`? Think `sep` would only be needed with `paste` function.

**Comment by millerg23** [comment]
Looks good @zdz2101 , works nicely. Just a minor comment from me.

File: R/period_dataset.R (hunk 1 above)
Comment (zdz2101) [reply R/period_dataset.R:190]: you're absolutely right, I'll remove, then ask a couple others to review then we can close?

**Comment by bms63** [comment]
We need a new entry for this one.  

Also - does a test need to be written for this?  It is listed as a bug

**Comment by zdz2101** [comment]
@bms63 looks ready now

**Pull request merged**